mobile/Ihc/services/LocalDataService.js

```diff
@@ -230,16 +233,37 @@ export function updateMedication(update) {
       properties.forEach( p => {
         medication[p] = update[p];
       });
-      return;
+      medication.key = Medication.makeKey(update);
```

Comment:
The main use of a key is that it is a unique identifier that doesn't change. I'd recommend that for a given drug, fields like the name, dose, and unit should stay constant, but the quantity can be changed. Thus you'd make the key a function of the name dose and unit. Otherwise, if we referenced this key somewhere else, but this key changes, then that reference becomes invalid

Reply:
What about in the case where some input is incorrectly submitted... Say they spelled the name of the drug wrong, etc...

Reply:
Well because of the key's role as a unique identifier, you either keep the name uneditable, or you produce keys a different way. Something else you could do is just have some counter variable stored, that increments every time a new drug is added, and use that as the key. If you did that, then the key would be in the schema of both the mobile and server databases, but you wouldn't show it in any UI. However that could be complicated because that counter would have to be stored on the server to make sure there aren't any conflicts between tablets... By far the simplest way for us is keep names uneditable, and if they must correct it, then just make another object and delete the old one

Reply:
what if we check to see if a medication already exists for a given key everytime a new medication is created/updated? And if it does, show a message saying that it already exists. This way, no two medications can have the same key.

Reply:
You should already be checking to see if a medication exists for a given key everytime. For creates, you should be checking it to make sure it doesn't already exist, and for updates you should be checking to make sure it does exist. It sounds like what you are suggesting is to just ban updates altogether? I don't think we can remove the update functionality, because then how would we adjust the quantity?

Reply:
Oh what I meant was when updating, check to see if the new key (i.e. after changing the name) already exists. And for the quantity, I think ayush implemented an increment/decrement button, but I guess we should still allow the user to update the number in the case of restocking.

Reply:
Oh the problem isn't conflicts with other keys, the problem is dealing with other references to the old key. I left another comment in the Server functions about how it will break if you change the keys, because the server won't know what the new key is, and even if you setup some custom logic in the server to update it, propagating that to all the tablets after will probably be rough. And increment/decrement is still updating the object, so it will require update calls to the local database and the server. Also as a side point, I don't think increment/decrement by 1 is the best way to deal with changing quantities... I think you should just have an input field where they can type in a number. A prescription of say 20 or 40 pills isn't uncommon, so we shouldnt make them push the button every time

Reply:
Ahh gotcha. I'll bring it up during our next meeting.